libraries/helpers.rb

```diff
@@ -57,14 +57,16 @@ def shell_is_valid?(shell_path)
     # Validates passed id.
     #
     # @return [Numeric, String]
-    # handles checking whether uid was specified as a string
+    # handles basic input validation and checking 
+    # whether uid was specified as a string
     def validate_id(id)
+      return if id.nil? || id == 0
       id.to_i.to_s == id ? id.to_i : id
```

Comment:
Also, I'm really not sure what this is supposed to achieve here, if we just need to make sure that id is an `integer` then you can just call `.to_i` on whatever is passed in.

Reply:
This should be for line 64

Reply:
Yeah most of the things you are commenting on were already in the code. I don't mind refactoring it though, so I will have a look at it :) 

I mean we could just kill the whole validate_id and just use .to_i in the resource? I mean we are already validating if the hash has a key called 'gid' or 'uid'.  The worst thing that would happen is that nil would become 0, which would just error out as you won't be able to overwrite root.

What do you think?

Reply:
I decided to go for it as it does not really make sense to have a method to check if something is an integer :) validate_id has been killed off.